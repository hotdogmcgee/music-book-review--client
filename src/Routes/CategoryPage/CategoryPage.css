.instrument-view-reminder {
    text-align: center;
}