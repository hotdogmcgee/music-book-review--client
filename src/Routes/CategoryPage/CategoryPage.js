import React from "react";
import { Link } from "react-router-dom";
import BookList from "../../Components/BookList/BookList";
import FilterSortBar from "../../Components/FilterSortBar/FilterSortBar";
import { Section } from "../../Components/Utils/Utils";
import { STORE } from "../../store.js";
import "./CategoryPage.css";

export default class CategoryPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      bookList: [],
      browseTrue: false,
      listSorted: false,
      sortVal: null
    };
    this.handleSortOption = this.handleSortOption.bind(this);
  }

  componentDidMount() {
    console.log(this.props.match.params);
    this.setState(
      {
        bookList: STORE.bookList
      },
      () => {
        //still need to include instrument filters, put this cb in a separate method

        // const instrument = this.props.match.params.instrument;
        // if (instrument) {
        //   this.handleSortOption(instrument);
        // }
        const type = this.props.match.params.type;
        if (type) {
          this.handleSortOption(type);
        }
      }
    );
  }

  //load list in app and do a cb func?

  //use context to access data
  //take conditional path as prop to determine what to show

  // static defaultProps = {
  //   category: "instrument",
  //   list: []
  // };

  // handleInstrumentCategory = () => {
  //   if (this.props.match.)
  // }

  handleSortOption(sortValue) {
    console.log("sortValue: ", sortValue);

    const list = this.state.bookList ? this.state.bookList : STORE.bookList;

    const sortFunc = function(a, b) {
      var thingA = a[sortValue].toUpperCase();
      var thingB = b[sortValue].toUpperCase();
      if (thingA < thingB) {
        return -1;
      }
      if (thingA > thingB) {
        return 1;
      }

      return 0;
    };

    let newList;

    if (sortValue !== this.state.sortVal) {
      switch (sortValue) {
        case "instrument":
          newList = list.sort(sortFunc);

          break;
        case "author":
          console.log("author");
          newList = list.sort(sortFunc);
          break;
        case "id":
          newList = list.sort((a, b) => {
            return a.id - b.id;
          });
          break;
        case "title":
          newList = list.sort(sortFunc);
          break;
        case "rating":
          newList = list.sort((a, b) => {
            return b.rating - a.rating;
          });
          break;
        default:
          console.log("yo");
      }

      this.setState({
        bookList: newList,
        listSorted: true,
        sortVal: sortValue
      });
    } else if (this.state.listSorted) {
      newList = list.reverse();

      this.setState({
        bookList: newList
      });
    }
  }

  handleFilterOption = filters => {
    //store big list in context
    //is it better to just hide an element based on filter?
    // const list = this.state.bookList ? this.state.bookList : STORE.bookList
    const list = STORE.bookList;

    let newList = list;

    if (!filters.includes("old books")) {
      newList = newList.filter(item => item.published_year <= 2000);
    }

    if (!filters.includes("under 25 dollars")) {
      newList = newList.filter(item => item.cost < 25);
    }
    if (!filters.includes("recent")) {
      newList = newList.filter(item => item.published_year > 2000);
    }

    this.setState({
      bookList: newList
    });
  };

  render() {
    return (
      <>
        <Section>
          <Link to="/">Music Book Review</Link>
        </Section>

        <FilterSortBar
          onSortOptionClick={this.handleSortOption}
          onFilterOptionClick={this.handleFilterOption}
        />

        <BookList bookList={this.state.bookList} />
      </>
    );
  }
}
