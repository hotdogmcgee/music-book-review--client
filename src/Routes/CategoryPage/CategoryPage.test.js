import React from "react";
import CategoryPage from "./CategoryPage";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("CategoryPage component", () => {
  it("renders a CategoryPage by default", () => {
      const context = {  filterObject: {
        searchValue: "",
        browseValue: "",
        instrumentValue: "",
        filterValue: "",
        sortValue: "",
        listSorted: false
      }}
    const wrapper = shallow(<CategoryPage />, { context }) ;
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});