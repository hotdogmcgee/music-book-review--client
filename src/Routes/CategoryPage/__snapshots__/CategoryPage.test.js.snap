// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPage component renders a CategoryPage by default 1`] = `
<Fragment>
  <FilterSortBar />
  <BookList
    bookList={Array []}
  />
</Fragment>
`;
